nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -286,7 +286,7 @@ let
     if serial; then set with_serial=yes ;fi
     export with_serial
     clear
-    set timeout=10
+    set timeout=${toString config.boot.loader.timeout}
```

Comment:
What happens when `config.boot.loader.timeout == null` which is a valid option meaning "infinite timeout"?

Reply:
Perhaps I should change to use similar logic as syslinuxTimeout (coercing `null` to `0`)? (https://github.com/NixOS/nixpkgs/blob/c4208e83e9146977762ce3185164a381a65434ce/nixos/modules/installer/cd-dvd/iso-image.nix#L74)

Reply:
it's -1 for infinite timeout in GRUB, 0 is a valid value for instantaneous wait

Reply:
but yes :)

Reply:
Thanks for your input. What do you think about 1a1e24d36cd4c7c33a319fbbec9f43d208eb67f9 ?

I'm not sure that `null` -> `0` makes total sense, I would think `null` -> `default` (10 secs) would make more sense, but probably more important that it be consistent with the syslinux timeout.

Reply:
I think, null should be "no timeout", so infinite wait IMHO, see docs about the option:
`Timeout (in seconds) until loader boots the default menu item. Use null if the loader menu should be displayed indefinitely.`